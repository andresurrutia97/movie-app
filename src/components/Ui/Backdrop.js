import React from 'react';
import PropTypes from 'prop-types';

const Backdrop = ({ show, closed }) => {
  const baseBackdropClass = 'fixed w-full h-full bg-black opacity-50 z-30';

  const backdropClass = show
    ? [baseBackdropClass, 'block']
    : [baseBackdropClass, 'hidden'];

  return (
    <div
      className={backdropClass.join(' ')}
      role="presentation"
      onClick={closed}
    />
  );
};

Backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  closed: PropTypes.func,
};

Backdrop.defaultProps = {
  closed: () => {},
};

export default Backdrop;
