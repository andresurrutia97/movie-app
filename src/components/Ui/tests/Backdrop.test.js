import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Backdrop from '../Backdrop';

describe('<Backdrop/>', () => {
  const closed = jest.fn();
  const component = <Backdrop show={false} closed={closed} />;

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(component);
  });

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call the close function when click an item', () => {
    wrapper.simulate('click');
    expect(closed).toBeCalledTimes(1);
  });

  it('should be a block element when show prop is true', () => {
    expect(wrapper.props().className).toMatch(new RegExp('hidden'));
    wrapper.setProps({ show: true });
    expect(wrapper.props().className).toMatch(new RegExp('block'));
  });
});
