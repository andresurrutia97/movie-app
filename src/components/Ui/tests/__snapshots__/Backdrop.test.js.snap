// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop/> should render correctly 1`] = `
<div
  className="fixed w-full h-full bg-black opacity-50 z-30 hidden"
  onClick={[MockFunction]}
  role="presentation"
/>
`;
